docs/Dealing-with-Duplication-of-Data.md

```diff
@@ -0,0 +1,108 @@
+
+
+## Processing configuration
+
+To get higher levels of abstraction it is necessary to process configuration files and data.
+
+One challenge is to correctly process configuration. In some cases the only reliable way how to do that is to use the native tools using the configuration. As a middle ground existing abstractions of parsers, such as Augeas or what is in YaST could be helpful.
+
+Another challenge is to where to process the data. In some cases it might be necessary to do that in the environment of the running system, in other cases, extracted data is enough.
+
+
+## Use cases
+
+Which level of abstraction is useful depends on the use case. Examples:
+
+### Configuration discovery
+
+When analysing a system, especially when trying to figure out what an unknown system does, all levels of abstraction are helpful. Even the same data on different levels can be helpful. The more data is available the more easy it is for the user to find and judge the configuration of the system.
+
+### Replication of a system
+
+When recovering a system or when replicating it in general, a higher abstraction is not necessary. Using raw data encapsulating the state of a system in its native form is the most reliable way to achieve an exact replication.
+
+### Migration
+
+For a migration use case it depends on what level of abstraction the migration is taking place. When it is a low-level migration such as migrating a physical to a virtual system, the use case is closer to a replication of a system. When it on a higher level, e.g. when moving a service from one operating system to another, the lower level configuration might simply not work because the config file format could have changed or even a different tool used which provides the same service (e.g. Lighttpd instead of Apache). But correct configuration can be deduced from the higher level representation.
+
+For all migration use cases it is important that the result is in a consistent state, so that the target system is functional in itself.
+
+### Editing a system description
+
+When modifying a system description manually or programmatically, it depends on the knowledge of the user, what abstraction level is most convenient. If a user knows the format of a configuration file, it might be easiest to edit that directly. In other cases it might be more convenient to edit the abstract description, so that it can be used in a more abstract way and for example be applied to different target systems.
+
+There can't be a clear preference here. The challenge is to avoid problems caused by inconsistencies of information on different levels.
+
+
+## Requirements
+
+In the context of the internal consistency of system description we have a number of requirements:
+
+* No data should be lost, unless explicitly desired by the user.
```

Comment:
I woudn't phrase it that strong. In certain use cases the user implicitly agrees on certain data to be lost.